altair/utils/save.py

```diff
@@ -9,28 +9,26 @@
 
 
 def write_file_or_filename(
-    fp: Union[str, pathlib.PurePath, IO],
+    fp: Union[pathlib.Path, IO],
     content: Union[str, bytes],
     mode: str = "w",
     encoding: Optional[str] = None,
 ) -> None:
     """Write content to fp, whether fp is a string, a pathlib Path or a
     file-like object"""
-    if isinstance(fp, str) or isinstance(fp, pathlib.PurePath):
+    if isinstance(fp, pathlib.Path):
```

Comment:
If you keep the original, I'd recommend adding [ruff SIM101](https://docs.astral.sh/ruff/rules/duplicate-isinstance-call/) to quick fix multiple isinstance calls

Reply:
Based on the comment above, let's keep the `str` isinstance check but I like the idea of adding SIM101! Could you add it to pyproject.toml and autofix everything it finds in a new commit?

Reply:
Thanks for that! Looks good. Personally, I like autofixable ruff rules which harmonize the code base so I'd be open to discuss this further. Thanks for bringing it up!